diag_manager2/fms_diag_axis.F90

```diff
@@ -284,4 +291,70 @@ integer function get_axis_set_num(set_name)
     end do
 end function get_axis_set_num
 
+
+
+
+!! Register and write the information of the axis.
+!! Assumes: axes object is initalized by this point
+!! TODO : It would be convenient to store axis info inside the axis object
+!!      so that conversions (e.g. character case) are not needed for writing or testing.
+!! TODO : Other file types besides FmsNetcdfFile_t
+!! REFERENCE: See functions diag_util.F90::opening_file and diag_output.F90::write_axis_meta_data
+!!   and write_field_meta_data
+subroutine send_data_generic(this, fobj)
+    class(diag_axis_type),   intent(inout)   :: this
+    type(FmsNetcdfFile_t),  intent(inout)   :: fobj
+    class(FmsNetcdfFile_t), pointer    :: fptr
+    !!
+    select type (fptr) !! Also check X or Y axis
+        type is (FmsNetcdfUnstructuredDomainFile_t)
+            call send_data_ncd_udft(this, fptr)
+        type is (FmsNetcdfDomainFile_t)
+            call send_data_ncd_dft(this, fptr)
+        class default
+            call error_mesg("diag_output_mod::write_axis_meta_data", &
+                   "The file object is not the right type. It must be FmsNetcdfDomainFile_t for a "//&
+                    "X or Y axis", FATAL)
+    end select
+end subroutine send_data_generic
+
+subroutine send_data_ncd_common(axis, fobj)
```

Comment:
These subroutines all need doxygen comments at the top describing what they do, and the variables need doxygen comments explaining what they are.